flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala

```diff
@@ -1225,16 +1225,23 @@ abstract class TableTestUtilBase(test: TableTestBase, isStreamingMode: Boolean)
     // check whether the sql equals to the expected if the `relNodes` are translated from sql
     assertSqlEqualsOrExpandFunc()
     // check ast plan
+    val plans = new util.HashSet[PlanKind](expectedPlans.toSeq.asJava)
+
     if (expectedPlans.contains(PlanKind.AST)) {
+      plans.remove(PlanKind.AST)
       assertEqualsOrExpand("ast", astPlan)
     }
     // check optimized rel plan
     if (expectedPlans.contains(PlanKind.OPT_REL)) {
-      assertEqualsOrExpand("optimized rel plan", optimizedRelPlan, expand = false)
+      plans.remove(PlanKind.OPT_REL)
+      // if there is something else in expected plans, then use it for xml generation if needed
+      assertEqualsOrExpand("optimized rel plan", optimizedRelPlan, !plans.isEmpty)
```

Comment:
There are some tests requiring both `OPT_REL`, `OPT_REL_WITH_ADVICE`
in order to have both generated in xml, need to put `true` here if both are present